Objects/tupleobject.c

```diff
@@ -76,25 +85,25 @@ _PyTuple_DebugMallocStats(FILE *out)
 #endif
 }
 
-PyObject *
-PyTuple_New(Py_ssize_t size)
+/* Return half baked tuple. Before making it public following steps must be
+   done:
+   - initialize its items
+   - call TUPLE_GC_TRACK() on it
+   Because the empty tuple is always reused and it's already tracked by GC,
+   this function must not be called with size == 0 (unless from PyTuple_New()
+   which wraps this function).
+*/
+static PyTupleObject *
+tuple_new_untracked(Py_ssize_t size)
```

Comment:
I think that I understood why I'm not fully comfortable with this PR: it's the "tuple_new_untracked" name. Usually "new" functions in Python return a fully initialized Python object, ready to be used. Here it's more or less "uninitialized raw memory". I suggest to rename the function to tuple_alloc() and replace "Return half baked tuple." with "Allocate an uninitialized tuple object." in the comment.

Reply:
done